lib/ansible/modules/cloud/google/gcspanner.py

```diff
@@ -0,0 +1,269 @@
+#!/usr/bin/python
+# Copyright 2017 Google Inc.
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+ANSIBLE_METADATA = {'status': ['preview'],
+                    'supported_by': 'community',
+                    'version': '1.0'}
+DOCUMENTATION = '''
+---
+module: gcspanner
+version_added: "2.3"
+short_description: Create and Delete Instances/Databases on Spanner.
+description:
+    - Create and Delete Instances/Databases on Spanner.
+      See U(https://cloud.google.com/spanner/docs) for an overview.
+requirements:
+  - "python >= 2.6"
+  - "google-auth >= 0.5.0"
+  - "google-cloud-spanner >= 0.23.0"
+notes:
+  - Changing the configuration on an existing instance is not supported.
+author:
+  - "Tom Melendez (@supertom) <tom@supertom.com>"
+options:
+  configuration:
+    description:
+       - Configuration the instance should use. Examples are us-central1, asia-east1 and europe-west1.
+    required: True
+  instance_id:
+    description:
+       - GCP spanner instance name.
+    required: True
+  database_name:
+    description:
+       - Name of database contained on the instance.
+    required: False
+  instance_display_name:
+    description:
+       - Name of Instance to display.  If not specified, instance_id will be used instead.
+    required: False
+  node_count:
+    description:
+       - Number of nodes in the instance.  If not specified while creating an instance,
+         node_count will be set to 1.
+    required: False
+  state:
```

Comment:
I'm not sure if this should be the behavior of this option. As-is, this module could delete an instance if someone passes `database_name=null` (or does so by accident with bad variable assignment). If I understand docs correctly, deleting an instance would delete *all* databases on that instance, which would be really unfortunate. 

Can you require a `force: yes` type argument to delete whole instances?